webapi/scripts/fuseki-init-knora-test.sh

```diff
@@ -0,0 +1,113 @@
+#!/usr/bin/env bash
+
+#set -x
+
+POSITIONAL=()
+while [[ $# -gt 0 ]]; do
+  key="$1"
+
+  case $key in
+  -u | --username)
+    USERNAME="$2"
+    shift # past argument
+    shift # past value
+    ;;
+  -p | --password)
+    PASSWORD="$2"
+    shift # past argument
+    shift # past value
+    ;;
+  -h | --host)
+    HOST="$2"
+    shift # past argument
+    shift # past value
+    ;;
+  *) # unknown option
+    POSITIONAL+=("$1") # save it in an array for later
+    shift              # past argument
+    ;;
+  esac
+done
+set -- "${POSITIONAL[@]}" # restore positional parameters
+
+FILE="$1"
+
+REPOSITORY="knora-test"
+
+if [[ -z "${HOST}" ]]; then
+  HOST="localhost:3030"
+fi
+
+if [[ -z "${USERNAME}" ]]; then
+  USERNAME="admin"
+fi
+
+if [[ -z "${PASSWORD}" ]]; then
+  PASSWORD="test"
+fi
+
+delete-repository() {
+  STATUS=$(curl -s -o /dev/null -w '%{http_code}' -u ${USERNAME}:${PASSWORD} -X DELETE http://${HOST}/\$/datasets/${REPOSITORY})
+
+  if [ "${STATUS}" -eq 200 ]; then
+    echo "==> delete repository done"
+    return 0
+  else
+    echo "==> delete repository failed"
+    return 1
+  fi
+}
+
+create-repository() {
+  STATUS=$(curl -s -o /dev/null -w '%{http_code}' -u ${USERNAME}:${PASSWORD} -F data=@./fuseki-knora-test-repository-config.ttl http://${HOST}/\$/datasets)
+
+  if [ "${STATUS}" -eq 200 ]; then
+    echo "==> create repository done"
+    return 0
+  else
+    echo "==> create repository failed"
+    return 1
+  fi
+}
+
+upload-graph() {
+  STATUS=$(curl -s -o /dev/null -w '%{http_code}' -u ${USERNAME}:${PASSWORD} -H "Content-Type:text/turtle" -d @$1 -X PUT http://${HOST}/${REPOSITORY}\?graph\="$2")
+
+  if [ "${STATUS}" -eq 201 ]; then
+    echo "==> 201 Created: $1 -> $2"
+    return 0
+  elif [ "${STATUS}" -eq 200 ]; then
+    echo "==> 200 OK: $1 -> $2"
+    return 0
+  else
+    echo "==> failed with status code ${STATUS}: $1 -> $2"
+    return 1
+  fi
+}
+
+# delete-repository // delete dos not work correctly. need to delete database manually.
```

Comment:
Why not send a SPARQL `DROP ALL` instead of deleting the repository?

Reply:
Init is used to initialize the repository, i.e., create a new repository based on the configuration. If we only empty the repository, then a new one cannot be created, because there is already one with that name.

The Admin API route for deleting the repository only deletes the TDB2 database but does not delete the Apache Lucene database, so when a new repository is created, an error message is returned. This is why the repository needs to be deleted by hand, or in our case through a special make target.

Reply:
Could we just ignore those errors?

Reply:
Never mind, I understand: you need to be able to recreate the repository with a different configuration.

Reply:
👍